src/List/List.js

```diff
@@ -36,11 +40,31 @@ const List = React.forwardRef(({
         className
     );
 
+    const isHeader = (child) => child?.type?.displayName === ListHeader.displayName;
+    const isFooter = (child) => child?.type?.displayName === ListFooter.displayName;
+
+    const listHeader = React.Children.toArray(children).find(isHeader);
+    const listFooter = React.Children.toArray(children).find(isFooter);
+    const listId = useUniqueId(id);
+
     return (
-        <ul {...props} className={ListClasses}
-            ref={ref}>
-            { React.Children.map(children, child => React.cloneElement(child, { hasByline, navigation, partialNavigation })) }
-        </ul>
+        <>
+            {listHeader && React.cloneElement(listHeader, { id: `${listId}-label` })}
+            <ul
+                {...props}
+                {...(selectable ? { 'role': 'listbox' } : null)}
+                aria-labelledby={listHeader ? `${listId}-label` : null}
+                className={ListClasses}
+                id={`${listId}-list`}
+                ref={ref}>
+                { React.Children.map(children, child => {
+                    if (!(isHeader(child) || isFooter(child)) ) {
```

Comment:
probably a nitpick
```js
if (!(isHeader(child) || isFooter(child)) ) {
vs
if (!isHeader(child) && !isFooter(child)) {
```
But it brings up a bigger issue to me, we aren't supporting custom components or fragments.  We **should** be able to consume this code and add the right props to the right children

```jsx
const MyOtherItems = () => (
   <React.Fragment>
        <List.Item>
            <List.Text>List Item 2</List.Text>
        </List.Item>
        <List.Item>
            <List.Text>List Item 3</List.Text>
        </List.Item>
        <List.Item>
            <List.Text>List Item 4</List.Text>
        </List.Item>
   </React.Fragment>
);
// ....
const MyListFooter = () => <List.Footer>List Footer</List.Footer>
// ...
<List>
        <List.Header>List Header</List.Header>
        <List.Item>
            <List.Text>List Item 1</List.Text>
        </List.Item>
        <MyOtherItems />
        <MyListFooter />
    </List>
```

Reply:
Thanks for pointing this out. I removed the children scanning and made the header and footer as List props.

Reply:
That's an interesting idea to get around the problem.  I think it would have to look more like this though
```jsx
<List
  footer='My footer'
  header='My header'>
   // ...
```
and the List component would render them like
```jsx
return (
  <>
    {header && <List.Header id={`${listId}-label`}>{header}</List.Header>}
    // ...
    {footer && <List.Footer>{footer}</List.Footer>}
  </>
```

Reply:
This will mean that we cannot get fancy with the header and footer [like here](https://github.com/SAP/fundamental-react/pull/1167/files#diff-e225e0a8c2fdc1854374ffe9ca6096deR150-R153), right?

Reply:
I guess it would look like
```jsx
footer = {<React.Fragment><List.Text><b>Total</b></List.Text><List.Text secondary>7</List.Text></React.Fragment>}
```
instead of
```jsx
        footer={
            <List.Footer>
                <List.Text><b>Total</b></List.Text>
                <List.Text secondary>7</List.Text>
            </List.Footer>
        }
```